src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -46,9 +54,16 @@ public TElement[] ToArray()
 
             TElement[] array = new TElement[count];
             int[] map = SortedMap(buffer);
```

Comment:
Yeah, replace all the current ones.

Return is definitely [unhappy if you pass it null](https://github.com/dotnet/corefx/blob/master/src/System.Buffers/src/System/Buffers/DefaultArrayPool.cs#L124)